language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToConfigurableCodeAction.java

```diff
@@ -0,0 +1,210 @@
+                        || q.text().equals(Qualifier.CONFIGURABLE.getValue()));
+    }
+
+    private static boolean isVariableAssigned(List<Location> references, NonTerminalNode rootNode) {
+        return references.stream()
+                .map(location -> rootNode.findNode(location.textRange()))
+                .anyMatch(node -> node.parent().kind() == SyntaxKind.ASSIGNMENT_STATEMENT);
+    }
+
+    private static boolean isListVarUpdated(List<Location> references, NonTerminalNode rootNode) {
+        for (Location location : references) {
+            NonTerminalNode node = rootNode.findNode(location.textRange());
+            NonTerminalNode parentNode = node.parent();
+            SyntaxKind parentKind = parentNode.kind();
+            if (parentKind == SyntaxKind.INDEXED_EXPRESSION
+                    && parentNode.parent().kind() == SyntaxKind.ASSIGNMENT_STATEMENT) {
+                return true;
+            }
+            if (parentKind == SyntaxKind.METHOD_CALL) {
+                if (LIST_UPDATING_METHODS.contains(((MethodCallExpressionNode) parentNode)
+                        .methodName().toSourceCode())) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private static boolean isTableVarUpdated(List<Location> references, NonTerminalNode rootNode) {
+        for (Location location : references) {
+            NonTerminalNode node = rootNode.findNode(location.textRange());
+            NonTerminalNode parentNode = node.parent();
+            if (parentNode.kind() == SyntaxKind.METHOD_CALL) {
+                if (TABLE_UPDATING_METHODS.contains(((MethodCallExpressionNode) parentNode)
+                        .methodName().toSourceCode())) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private static boolean isMappingVarUpdated(List<Location> references, NonTerminalNode rootNode) {
+        for (Location location : references) {
+            NonTerminalNode node = rootNode.findNode(location.textRange());
+            if (node.kind() == SyntaxKind.ASSIGNMENT_STATEMENT) {
+                ExpressionNode exprNode = ((AssignmentStatementNode) node).expression();
+                if (exprNode.kind() == SyntaxKind.METHOD_CALL) {
+                    MethodCallExpressionNode methodCallExpressionNode = (MethodCallExpressionNode) exprNode;
+                    return MAPPING_UPDATING_METHODS.contains(methodCallExpressionNode.methodName().toSourceCode());
+                }
+                return false;
+            }
+            NonTerminalNode parentNode = node.parent();
+            if (parentNode.kind() == SyntaxKind.INDEXED_EXPRESSION
+                    && parentNode.parent().kind() == SyntaxKind.ASSIGNMENT_STATEMENT) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
```

Comment:
This is the only place we use `NAME` right?

Reply:
yes.

Reply:
May be this also done as the previous code action have it. But I think we don't need to do it since we already take the name from the `getName` function.

Reply:
Updated.